### PR TITLE
rgw: dump ETag header for PostObj

### PR DESCRIPTION
**Pull request by Rjerk**
Signed-off-by: Liu Lan <liulan_yewu@cmss.chinamobile.com>

ref: https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -3112,6 +3112,7 @@ void RGWPostObj_ObjStore_S3::send_response()
   dump_errno(s);
   if (op_ret >= 0) {
     dump_content_length(s, s->formatter->get_len());
+    dump_header_if_nonempty(s, "ETag", etag);
```
Comment (dang) [inline src/rgw/rgw_rest_s3.cc:3115]: Isn't etag already dumped a 3107?

Comment (Rjerk) [reply src/rgw/rgw_rest_s3.cc:3115]: 3107 means?

Comment (tchaikov) [reply src/rgw/rgw_rest_s3.cc:3115]: https://github.com/ceph/ceph/blob/9809db9099d13986454e2f8279ab06a62a73131e/src/rgw/rgw_rest_s3.cc#L3107

Comment (Rjerk) [reply src/rgw/rgw_rest_s3.cc:3115]: it dumped in the body. 

I found it also dumped in headers in AWS S3: https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html

```
HTTP/1.1 100 Continue
HTTP/1.1 200 OK
x-amz-id-2: LriYPLdmOdAiIfgSm/F1YsViT1LW94/xUQxMsF7xiEb1a0wiIOIxl+zbwZ163pt7
x-amz-request-id: 0A49CE4060975EAC
x-amz-version-id: default
Date: Wed, 12 Oct 2009 17:50:00 GMT
ETag: "1b2cf535f27731c974343645a3985328"
Content-Length: 0
Connection: close
Server: AmazonS3
```

**Comment by Rjerk** [comment]
ping

**Comment by tchaikov** [comment]
@Rjerk i am not qualified to review this change. but your commit message fails to explain why we need this change. i am not sure if https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html implies the "ETag" is mandatory in the header of response.

right the "Etag" is included in the samples included by https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html . but as Daniel put, it's also included in the body as required by https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html . does this fix address any test failure or address any use case?

**Pull request closed without merging**